src/vm/peimage.cpp

```diff
@@ -909,73 +909,149 @@ PTR_PEImageLayout PEImage::GetLayoutInternal(DWORD imageLayoutMask,DWORD flags)
     {
         _ASSERTE(HasID());
 
-        if (imageLayoutMask&PEImageLayout::LAYOUT_MAPPED)
+        BOOL bIsMappedLayoutSuitable;
+        BOOL bIsFlatLayoutSuitable;
+
+#ifdef PLATFORM_UNIX
+        bIsMappedLayoutSuitable = ((imageLayoutMask & PEImageLayout::LAYOUT_MAPPED) != 0);
```

Comment:
Since this is common to both platforms, the statement can be moved out to be earlier and used in the initialization of the variable.